codeface/R/cluster/graph_comparison.r

```diff
@@ -142,37 +173,134 @@ graph.comparison <- function(g.1, g.2) {
   g.1.intersect <- graph.adjacency(adj.matrix.1.intersect, mode = "directed")
   g.2.intersect <- graph.adjacency(adj.matrix.2.intersect, mode = "directed")
 
-  graph.diff <- graph.difference(g.1.intersect, g.2.intersect)
+  graph.diff <- graph.difference(g.1.intersect, g.2.intersect, weighted = weighted, symmetric = symmetric)
 
   return(graph.diff)
 }
 
+get.index.map1 <- function(ids) {
+  N        <- nrow(ids)
+  map      <- new.env(size=N)
+  for(i in 1:N) {
+    map[[as.character(ids$id1[i])]] <- ids$id[i]
+  }
+  return(map)
+}
 
+get.index.map2 <- function(ids) {
+  N        <- nrow(ids)
+  map      <- new.env(size=N)
+  for(i in 1:N) {
+    map[[as.character(ids$id2[i])]] <- ids$id[i]
+  }
+  return(map)
+}
 ################################################################################
 ## High Level Functions
 ################################################################################
-run.graph.comparison <- function(con, pid.1, range.id.1, pid.2, range.id.2) {
+run.graph.comparison <- function(con, range.id.1, range.id.2, weighted=FALSE, symmetric=FALSE) {
   ## TODO: cluster method should not actually be required, we only need the main
   ##       graph, unfortunatley the database scheme currently marries the graph
   ##       to a clustering method
   cluster.method="Spin Glass Community"
+  pid.1 <- get.project.id.from.release.range.id(con, range.id.1)
+  pid.2 <- get.project.id.from.release.range.id(con, range.id.2)
   graph.data.1 <- get.graph.data.local(con, pid.1, range.id.1, cluster.method)
   graph.data.2 <- get.graph.data.local(con, pid.2, range.id.2, cluster.method)
-  edgelist.1   <- graph.data.1$edgelist
-  edgelist.2   <- graph.data.2$edgelist
+  # get the raw edgelists with the ids from the database
+  edgelist.1   <- graph.data.1$edgelist.db
+  edgelist.2   <- graph.data.2$edgelist.db
```

Comment:
In this case it probably makes more sense to keep the edgelist.db (raw database ids) and remove the copied edgelist with magic local ids?